File: src/main/java/com/team04/back/domain/weather/weather/repository/WeatherRepository.java
package com.team04.back.domain.weather.weather.repository;

import com.team04.back.domain.weather.weather.entity.WeatherInfo;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.time.LocalDate;
import java.util.List;
import java.util.Optional;

@Repository
public interface WeatherRepository extends JpaRepository<WeatherInfo, Integer> {
    List<WeatherInfo> findByLocationAndDateBetween(String location, LocalDate start, LocalDate end);
    Optional<WeatherInfo> findByLocationAndDate(String location, LocalDate date);
}

File: src/main/java/com/team04/back/domain/weather/weather/repository/WeatherRepository.kt
package com.team04.back.domain.weather.weather.repository

import com.team04.back.domain.weather.weather.entity.WeatherInfo
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.time.LocalDate

@Repository
interface WeatherRepository : JpaRepository<WeatherInfo, Long> {
    fun findByLocationAndDateBetween(location: String, start: LocalDate, end: LocalDate): List<WeatherInfo>
    fun findByLocationAndDate(location: String, date: LocalDate): WeatherInfo?
}
